### PR TITLE
[IMP] orm: update changelog for _flush_search()

### PR DESCRIPTION
**Pull request by ryv-odoo**
https://github.com/odoo/odoo/pull/144747
https://github.com/odoo/enterprise/pull/53380
https://github.com/odoo/upgrade/pull/5469

### CODE REVIEW

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

File: content/developer/reference/backend/orm/changelog.rst [hunk 1]
```diff
@@ -4,6 +4,12 @@
 Changelog
 =========
 
+Odoo Online version 17.1
+========================
+
+- Method :meth:`~odoo.models.Model._flush_search` has been deprecated with
+  `#144747 <https://github.com/odoo/odoo/pull/144747>`_.
```
[the remark this answers is not in the document]
Comment (ryv-odoo) [reply content/developer/reference/backend/orm/changelog.rst:11]: I added references to the method and the object handling the flushing mechanism but I won't explain the whole PR here.

File: content/developer/reference/backend/orm/changelog.rst [hunk 2]
```diff
@@ -4,6 +4,13 @@
...
+  See `odoo.api.Environment.execute_query` and :class:`~odoo.tools.SQL`.
```
Comment (rco-odoo) [inline content/developer/reference/backend/orm/changelog.rst:12]:
```suggestion
  The flushing of fields is now done by :meth:`~odoo.api.Environment.execute_query`,
  and is based on metadata put in the :class:`~odoo.tools.SQL` object by
  :meth:`~odoo.models.BaseModel._search` and other low-level ORM methods that
  build such objects.  Those methods are also responsible for checking the access
  rights on the fields that are used in the SQL object.
```

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+

**Comment by ryv-odoo** [comment]
@robodoo r+

**Pull request closed without merging**